src/debug-generator/debug-generator.c

```diff
@@ -142,6 +159,32 @@ static int generate_wants_symlinks(void) {
         return r;
 }
 
+static int install_debug_shell_dropin(const char *tty) {
+        const char *dropin = "/run/systemd/system/debug-shell.service.d/50-tty.conf";
+        const char *dir = "/run/systemd/system/debug-shell.service.d/";
+        _cleanup_free_ char *text = NULL;
+        int r;
+
+        r = mkdir_p(dir, 0755);
+        if (r < 0 && r != -EEXIST)
+                return log_debug_errno(r, "Failed to create directory '%s': %m", dir);
```

Comment:
in cases like this where the following write_string_file() would fail anyway if the mkdir didn't succeed we usually just cast the result of the mkdir_parents() to `(void)` and let the write_string_file() deal with it, but that doesn't matter